src/core/server/http/cookie_session_storage.ts

```diff
@@ -29,10 +31,10 @@ export interface SessionStorageCookieOptions<T> {
 }
 
 class ScopedCookieSessionStorage<T extends Record<string, any>> implements SessionStorage<T> {
-  constructor(private readonly server: Server, private readonly request: Request) {}
+  constructor(private readonly server: Server, private readonly request: Readonly<Request>) {}
```

Comment:
issue #33775 has point to restrict access to hapi `Request` object in `registerAuth` handler. thus we will operate `KibanaRequest`-ish object here